src/type_hash.rs

```diff
@@ -41,25 +39,11 @@ pub fn encode_type<T: StructType>(value: &T) -> String {
     buffer
 }
 
-lazy_static! {
```

Comment:
I am concerned about the performance implications of this. I'm also unsure whether "dynamic typing" is within the spirit of EIP-712. Note from the spec: [https://eips.ethereum.org/EIPS/eip-712#rationale-for-typehash](https://eips.ethereum.org/EIPS/eip-712#rationale-for-typehash)

> The typeHash is designed to turn into a compile time constant in Solidity

Reply:
I'm not yet saying "no" to this, just trying to think it through. EIP-712 is cryptography, and errors can be subtle. Is there a way we can expose caching a type hash to the consumer for performance?